backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -0,0 +1,139 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from datetime import datetime
+import flickrapi
+from backend_jobs.ingestion_pipeline.pipeline_lib.image_provider_interface import ImageProvider
+from backend_jobs.ingestion_pipeline.pipeline_lib.data_types import ImageType
+from backend_jobs.ingestion_pipeline.pipeline_lib.data_types import ImageAttributes
+from backend_jobs.pipeline_utils import constants
+
+_FLICKR_API_KEY = '2d00397e012c30ccc33ca4fdc05a5c98'
+_FLICKR_SECRET_KEY =  'e36a277c77f09fdd'
+
+class FlickrProvider(ImageProvider):
+    """ This class is an implementation for the ImageProvider interface.
+    """
+    def __init__(self, query_arguments=None):
+        if query_arguments is not None:
+            self.query_arguments={'tag': query_arguments}
+
+    def get_images(self, num_of_page):
+        flickr = flickrapi.FlickrAPI(_FLICKR_API_KEY, _FLICKR_SECRET_KEY, cache = True)
+        photos = flickr.photos.search(text=self.query_arguments['tag'],
```

Comment:
why filtering on both text and tag? worthy of comment

Reply:
added comment